indico/modules/events/forms.py

```diff
@@ -135,3 +136,42 @@ class UnlistedEventsForm(IndicoForm):
     authorized_creators = PrincipalListField(_('Authorized users'), [HiddenUnless('enabled', preserve_data=True)],
                                              allow_external_users=True, allow_groups=True,
                                              description=_('These users may create unlisted events.'))
+
+
+class DataRetentionSettingsForm(IndicoForm):
+    minimum_data_retention = TimeDeltaField(_('Minimum data retention period'), units=('weeks',),
+                                            description=_('Specify the minimum data retention period (in weeks) '
+                                                          'configurable for registrations. This includes the '
+                                                          'retention period of individual registration form fields.'),
+                                            render_kw={'placeholder': _('Indefinite'), 'min': 1, 'max': 521})
+    maximum_data_retention = TimeDeltaField(_('Maximum data retention period'), units=('weeks',),
+                                            description=_(
+        'Specify the maximum data retention period (in weeks) configurable for registrations. This includes the '
+        'retention period of individual registration form fields. Note that setting this value will make the '
+        'retention period field mandatory during registration form setup.'),
+        render_kw={'placeholder': _('Indefinite'), 'min': 1, 'max': 521})
```

Comment:
This wrapping is really weird...

```suggestion
    maximum_data_retention = TimeDeltaField(
        _('Maximum data retention period'),
        units=('weeks',),
        description=_(
            'Specify the maximum data retention period (in weeks) configurable for registrations. This includes the '
            'retention period of individual registration form fields. Note that setting this value will make the '
            'retention period field mandatory during registration form setup.'
        ),
        render_kw={'placeholder': _('Indefinite'), 'min': 1, 'max': 521},
    )
```